src/piggui.rs

```diff
@@ -249,6 +439,7 @@ impl Application for Piggui {
     fn subscription(&self) -> Subscription<Message> {
         let subscriptions = vec![
             iced::event::listen().map(WindowEvent),
+            event::listen().map(ModalKeyEvent),
```

Comment:
We only need to add this subscription while the dialog is open, no?
If so, I would do like I do elsewhere, which is makde subscriptions a mut vec.... then add it only if show_modal == true.